### PR TITLE
Add editorconfig file

### LINKED ISSUE #141: Set up an editorconfig file
We need a `.editorconfig` file in order to ensure that the contributors use the same indentation settings as we do.

### PR DESCRIPTION
**Pull request by Alecaddd**
Implements the `.editorconfig` file so we can all code with more sanity and piece of mind.

- Fixes #141

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+root = true
+
+[*]
+end_of_line = lf
+indent_style = space
+trim_trailing_whitespace = true
+insert_final_newline = true
+charset = utf-8
+
+[*.{txt,build,json,xml,xml.in,in,py}]
```
Comment (bilelmoussaoui) [inline .editorconfig]: As a python dev, we would like to use 4 spaces here :P

Comment (Alecaddd) [reply .editorconfig]: That's fair :D

**Comment by albfan** [comment]
I did some nasty changes. Tell me your opinion @bilelmoussaoui and @Alecaddd  (doesn't need to be that way):

- Apply space indent in all vala files
- Add uncrustify config to format vala files automatically (and check them)
- Add a hacking guide

**Comment by Alecaddd** [comment]
I like this, thanks @albfan.
A couple of things tho.
* Please, revert the `reformat with uncrustify` commit and create a dedicated PR for it after we merge the editorconfig. I'd prefer to have them separate and not have a PR with 46 files that doesn't match the title and scope.
* I'm not a fan of spaces inside conditions: `if ( condition )` vs `if (condition)`.
* I think we should avoid requesting the space before the parenthesis when dealing with translatable strings: `_ ("string")`.

Thoughts?

**Comment by bilelmoussaoui** [comment]
> I like this, thanks @albfan.
> A couple of things tho.
> 
> * Please, revert the `reformat with uncrustify` commit and create a dedicated PR for it after we merge the editorconfig. I'd prefer to have them separate and not have a PR with 46 files that doesn't match the title and scope.
> * I'm not a fan of spaces inside conditions: `if ( condition )` vs `if (condition)`.
I prefer the later too.
> * I think we should avoid requesting the space before the parenthesis when dealing with translatable strings: `_ ("string")`.
Yeah totally agree about that! 
> 
> Thoughts?

**Comment by albfan** [comment]
Let me rework that. I can adjust uncrustify to match those expectations

**Comment by Alecaddd** [comment]
This looks better, but can you revert the edits to the python files, the debian control, and vala files?
Let's just merge this PR with the editorconfig, the uncrustify, and hack file.

**Comment by albfan** [comment]
I was puzzled by a setup that didn't match how our files are actually, but we can discuss that later.

**Pull request merged**